src/main/java/org/toradocu/extractor/Tag.java

```diff
@@ -1,56 +1,159 @@
 package org.toradocu.extractor;
 
-import java.util.Optional;
+import java.util.Objects;
+import org.toradocu.translator.spec.Specification;
+import org.toradocu.util.Checks;
 
-public interface Tag {
+/** Represents a Javadoc tag. Supported tags are registered in {@code Tag.Kind}. */
+public abstract class Tag<S extends Specification> {
 
-  enum Kind {
-    THROWS,
-    PARAM,
-    RETURN;
+  public enum Kind {
+    PARAM, // @param
+    RETURN, // @return
+    THROWS; // @throws and @exception
 
     @Override
     public String toString() {
-      switch (name()) {
-        case "THROWS":
-          return "@throws";
+      final String label = name();
+      switch (label) {
         case "PARAM":
           return "@param";
         case "RETURN":
           return "@return";
+        case "THROWS":
+          return "@throws";
         default:
-          throw new IllegalStateException("The value " + name() + " has no string representation.");
+          throw new IllegalStateException("The value " + label + " has no string representation.");
       }
     }
   }
 
+  /** The comment of this tag. */
+  private Comment comment;
+
+  /** The kind of this tag (e.g., @throws, @param). */
+  private final Kind kind;
```

Comment:
The order of fields is unexpected.  I would expect `kind` to come before `comment`.
Also, the order of getter methods in the class body is inconsistent with both orders.  The code is easier to read and review if fields are always listed and processed in a consistent order.